controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -738,14 +738,13 @@ public void setRequestEventWriter(EventStreamWriter<ControllerEvent> requestEven
                                                     } else {
                                                         return CompletableFuture.completedFuture(state);
                                                     }
-                                                }));
-                                    }, executor)
+                                                }))
                                             .thenApply(z -> status);
                                 });
                     } else {
                         return CompletableFuture.completedFuture(status);
                     }
-                }, executor)
+                }), RetryHelper.RETRYABLE_PREDICATE.or(e -> Exceptions.unwrap(e) instanceof RetriesExhaustedException), Integer.MAX_VALUE, executor)
```

Comment:
Shivesh, I did not understand why we would need Integer.MAX_VALUE retries. Wondering if there a better way instead of retrying it infinitely. Do share you thoughts on this.

Reply:
so we follow this pattern elsewhere too when we intend to add indefinite retries. Integer.Max = 2 billion.. lets say we retry once every second then this is = ~60+ years.. Which is practically indefinite retry.

Reply:
My question was more on the lines of a "fail fast" principle.  The approach in the PR would cause the controller to retry this operation multiple times without us knowing why it failed for so long. Perhaps there is really something wrong in the deployment and it will not evident to the users of Pravega...

Reply:
the users will fail fast -- both rest and grpc apis have timeouts. So their calls would timeout while controller would continute to retry since it had accepted the create request and started work on it.